docs/src/guide/systems_scheduler.md

```diff
@@ -0,0 +1,154 @@
++++
+type = "docs"
+title = "Systems and the scheduler"
+weight = 60
++++
+
+A key feature of entity-component systems is that 
+operations on the entities are organized in systems,
+which operate independently from one another and can
+be added or removed according to the respective needs.
+
+## Systems
+
+Systems can be thought of as functions that take a 
+{{< api World >}} instance and perform operations on the 
+world's entities. However, to allow storing intermediate
+variables between multiple system calls, and to 
+support special initializing and finalizing operations
+called at the beginning or the end of an ECS run, respectively,
+systems are expressed as structs implementing the 
+{{< api System >}} trait. This trait requires the
+systems to implement {{< api System.initialize initialize >}}, and
+{{< api System.finalize finalize >}} methods, called before or
+after the ECS run, respectively, and an {{< api System.update update >}} 
+method, called at every step of the ECS run. 
+Each of these methods takes a {{< api World >}} instance 
+on which they perform the desired operations. 
+
+```mojo {doctest="guide_systems_scheduler" global=true}
+from larecs import World, System
+
+@value
+struct Move(System):
+    
+    # This is executed once at the beginning
+    fn initialize(mut self, mut world: World) raises:
+        # We do not need to do anything here
+        pass
+
+    # This is executed in each step
+    fn update(mut self, mut world: World) raises:
+
+        # Move all entities with a position and velocity
+        for entity in world.query[Position, Velocity]():
+            entity.get[Position]().x += entity.get[Velocity]().dx
+            entity.get[Position]().y += entity.get[Velocity]().dy
+
+    # This is executed at the end
+    fn finalize(mut self, mut world: World) raises:
+        # We do not need to do anything here
+        pass
+```
+
+## Scheduler
+
+The {{< api Scheduler >}} is responsible for executing the systems
+in the correct order. A `Scheduler` contains a {{< api World >}} instance
+and a list of systems. The systems are executed in the order they
+are added to the scheduler. The scheduler has 
+{{< api Scheduler.initialize initialize >}},
+{{< api Scheduler.update update >}}, and {{< api Scheduler.finalize finalize >}} 
+methods, which call the respective functions of all
+considered systems in the order they are added to the scheduler. 
+In addition, the scheduler has a {{< api Scheduler.run run >}}
+method, which initializes the systems, runs them a desired 
+number of times, and finalizes them. 
```

Comment:
Not related to the docs, but still...

Simply running for a given number of steps my not be sufficient. E.g., a simulation may be terminated when the simulated population goes extinct. A game may run indefinitely until the user quits, wins or looses.

In Ark (or rather, `ark-tools`), I do this via a resource which contains a bool that systems can set to signal that the run can terminate.

So I think you will need some way to allow systems to terminate a (potentially infinite) run.